Add new 177 hooks, along with beginning of the version updates (#479)
PrestaShop/docs: src/content/1.7/modules/concepts/hooks/list-of-hooks.md

* `displayProductButtons` has been renamed into `displayProductAdditionalInfo`.<br>
  Don’t worry, we kept an alias :)

> Some hooks have been modified in the order view page in version {{< minver v="1.7.7" >}} see [Modified order hooks][modified-order-hooks] for more details.

Reviewer: I think this whole block can be removed, actually. All the detailed information should be noted in the "core updates" section.

Author: They are in the core updates page, I actually wanted to use this link in related hooks so that users can find more details easily But since the whole hook display relies on a `{{% funcdef %}}` I can't integrate links to the hook description.. So that's the best I found to keep a link available in this page
I'm not sure everyone knows about these core update pages, especially when they are just looking for info, but if you think they are enough I'll remove this block

Reviewer: Oh of course you can add links within a `funcdef` block. You just can't put references outside it:

```
{{% someshortcode %}}
[This link works]({{< ref "something.md" >}})
{{% /someshortcode %}}

{{% someshortcode %}}
[This link doesn't][some-link]
{{% /someshortcode %}}

[some-link]{{< ref "something.md" >}}
``

    );
    ```
    

> actionGetAdminOrderButtons

Reviewer: This one isn't new in 1.7.7?

Author: Yes it is The `available since` part is missing indeed, I'm gonna add it

: 

>     This hook is used to generate the buttons collection on the order view page thanks (see ActionsBarButtonsCollection)

Reviewer 2: ```suggestion
    This hook is used to generate the buttons collection on the order view page (see ActionsBarButtonsCollection)
```

Author: Damn, this means the core needs to be fixes as well 😅

    Located in: /src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

    Parameters:
    ```php
    array(
     'controller' => (OrderController) Symfony controller,
     'id_order' => (int) Order ID,
     'actions_bar_buttons_collection' => (ActionsBarButtonsCollection) Collection of ActionsBarButtonInterface
    );
    ```

actionAdminAdminPreferencesControllerPostProcessBefore
: 
    Available since: {{< minver v="1.7.7" >}}

>     On post-process in Admin Preferences', 'This hook is called on Admin Preferences post-process before processing the form

Reviewer 2: ```suggestion
    This hook is called on Admin Preferences post-process before processing the form
```

    );
    ```
    
actionGetAdminOrderButtons
: 
    Available since: {{< minver v="1.7.7" >}}

    This hook is used to generate the buttons collection on the order view page (see ActionsBarButtonsCollection)

    Located in: /src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

    Parameters:
    ```php
    array(
     'controller' => (OrderController) Symfony controller,
     'id_order' => (int) Order ID,
     'actions_bar_buttons_collection' => (ActionsBarButtonsCollection) Collection of ActionsBarButtonInterface

>     );

Reviewer: ```suggestion
    array(
       'controller' => (OrderController) Symfony controller,
       'id_order' => (int) Order ID,
       'actions_bar_buttons_collection' => (ActionsBarButtonsCollection) Collection of ActionsBarButtonInterface
    );
```

    Located in: /controllers/admin/AdminOrdersController.php

>     Replaced by **displayAdminOrderTabContent** in 1.7.7

Reviewer: I think it's probably best to note this at the top of the description instead at the bottom.

See line 303 for an example of how to do it.

    → `displayAdminOrderSide`

    Located in: admin-dev/themes/default/template/controllers/orders/helpers/view/view.tpl

displayAdminOrderSide
: 
    Available since: {{< minver v="1.7.7" >}}

    This hook displays content in the order view page at the end of the side column

    Located in: /src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig

    Parameters:
    ```php
    array(
      'id_order' => (int) Order ID
    );
    ```

displayAdminOrderSide
: 
    Available since: {{< minver v="1.7.7" >}}

    This hook displays content in the order view page in the side column under the customer view

    Located in: /src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig

    Parameters:
    ```php
    array(
      'id_order' => (int) Order ID
    );
    ```

displayAdminOrderSideBottom
: 
    Available since: {{< minver v="1.7.7" >}}

    This hook displays content in the order view page at the bottom of the side column

    Located in: /src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig

    Parameters:
    ```php
    array(
      'id_order' => (int) Order ID
    );
    ```

displayAdminOrderTabOrder
: 

>     **(removed since 1.7.7 in favor of)**

Reviewer: ```suggestion
    **(removed in 1.7.7)**
```
I think this works better

    
displayBackOfficeOrderActions
: 
    **(deprecated since 1.7.7)**
    → `actionGetAdminOrderButtons`

>     This hook displays content in the order view page after action buttons (or aliased to side column in migrated page)

Reviewer: > (or aliased to side column in migrated page)

I honestly don't understand this part 😅

Author: This one is a bit special:
- since 1.7.7 it is aliased to `displayAdminOrderSide` to avoid losing all the former actions, but it is not displayed in the same part on the page
- but the recommended way is to use the new `actionGetAdminOrderButtons` which is cleaner and adds action buttons at the top o the view

The alias feature was decided to keep a bit a backward retro compatibility but still motivate developers to migrate their module if they want their buttons to be on top of the page I'm not much of a fan of this, I think purely replace the hook would be much more efficient to motivate them ^^ (as it was done on other hooks)

Reviewer: I think that part must be clarified in the text.

    - themes/classic/templates/checkout/checkout.tpl
    - themes/classic/templates/layouts/layout-both-columns.tpl

displayAdditionalCustomerAddressFields
: 
    Available since: {{< minver v="1.7.7" >}}

>     This hook allows to display the extra field values added in an address from using hook 'additionalCustomerAddressFields'

Reviewer 2: ```suggestion
    This hook allows to display extra field values added in an address form using hook 'additionalCustomerAddressFields'
```
